### ISSUE TITLE
Make info command return ServerState struct

### ISSUE THREAD

**Issue by rustworthy** [jonhoo/faktory-rs#60]
Info command will now return a result with `serde_json::Value` which made a lot of sense during the period when the `Faktory` library was being actively developed and so the shape of the return type could change breakingly. The last time when that [interface](https://github.com/contribsys/faktory/blob/main/server/server.go#L433-L452) was updated was a couple of years [ago](https://github.com/contribsys/faktory/commit/4a57c90a844b6cd448cbc51cc14c7462e0de3539) and so we may want to introduce a dedicated type for it to make `Client::info` more ergonomic:

```rust
let server_state = client.info().await.unwrap(); // ServerState
let queues = &server_state.faktory.queues; // &HashMap<String, u64>
```

While currently - provided they know the keys - this will look like:
```rust
let server_state = client.info().await.unwrap(); // serde_json::Value
let queues = server_state.get("faktory").unwrap().get("queues").unwrap(); // &serde_json::Value
```

An example usage of `Client::info` can be found [here](https://github.com/jonhoo/faktory-rs/pull/59/files#diff-a07cf223951f9e59e9ab517b9c9b35ee99fc360ab0616a13aac62ffa2347143bR224-R225)

Since a breaking change is occurring due to switch to async, the `Client::info` can be updated to return `ServerState`.

**Comment by jonhoo** [comment]
That seems reasonable! If we also annotate it with `#[non_exhaustive]`, we'll even be able to expose more fields if they should be added.

**Comment by mperham** [comment]
Related, see my recent Faktory PR which adds a typed structure for INFO.

**Comment by jonhoo** [comment]
Ah, this is https://github.com/contribsys/faktory/pull/474?

Given we're doing a breaking change as part of #49 anyway, I think we should probably just remove `.info` entirely and add `.current_info` that returns a type matching the one you added here:
https://github.com/contribsys/faktory/pull/474/files#diff-4bbbde2a77215c352f9e99f0f2764fb9fb09eaa9052433b58a474b33019f7486R9

**Comment by rustworthy** [comment]
> Ah, this is [contribsys/faktory#474](https://github.com/contribsys/faktory/pull/474)?
> 
> Given we're doing a breaking change as part of #49 anyway, I think we should probably just remove `.info` entirely and add `.current_info` that returns a type matching the one you added here: https://github.com/contribsys/faktory/pull/474/files#diff-4bbbde2a77215c352f9e99f0f2764fb9fb09eaa9052433b58a474b33019f7486R9

Addressed in #59
